sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/ScriptTransformationExec.scala

```diff
@@ -161,6 +175,7 @@ case class ScriptTransformationExec(
 
               if (scriptOutputReader != null) {
                 if (scriptOutputReader.next(scriptOutputWritable) <= 0) {
+                  waitForTransformationProcessTermination()
                   checkFailureAndPropagate()
```

Comment:
https://github.com/apache/spark/blob/a6098ae0a89ff9f338ea138e0c17bc439e6e3a87/sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/ScriptTransformationExec.scala#L191

Here we call `proc.waitFor()`, shall we replace it with `waitForTransformationProcessTermination`?

Reply:
Sure